appservice/src/TunnelProxy.ts

```diff
@@ -115,6 +115,31 @@ class TunnelSocket extends EventEmitter {
     }
 }
 
+/**
+ * Interface for tunnel GetStatus API
+ */
+enum WebAppState {
+    STARTED = 'STARTED',
+    STARTING = 'STARTING',
+    STOPPED = 'STOPPED'
+}
+
+interface ITunnelStatus {
+    port: number;
+    canReachPort: boolean;
+    state: WebAppState;
+    msg: string;
+}
+
+/**
+ * Internal error indicating that we should continue to retry getting the tunnel status
+ */
+class RetryableTunnelStatusError extends Error {
+    constructor(message: string) {
```

Comment:
nit: This constructor is redundant and can be removed

Reply:
Good catch